src/gui/mzroll/mainwindow.cpp

```diff
@@ -908,6 +908,11 @@ void MainWindow::autosaveProject()
 
 void MainWindow::explicitSave()
 {
+    // the user is ordering an explicit save, get rid of autosave states
```

Comment:
@saifulbkhan If there is a crash during the explicit save process, we might lose all data. Would be safer to delete timestamp files once the explicit save is done successfully. Thoughts?

Reply:
Good catch. There will always be a "code-distance" between scrapping of timestamp file and saving of a new file (running in the save thread) that can fail and leave the user with neither of the files on their system. But we can minimize this for an apparent fail-safe process. Will make some changes and push soon.

Reply:
It turns out this here is more or less the optimum place where we should reset before explicit save.

Reply:
Is that because the currentProjectName is updated during explicitSave?

Reply:
Yes. And if you call the `resetAutosave` method after initiating a threaded save, the behavior would be undefined since `_currentProjectName` is used by the worker thread.

Reply:
Got it. Doesn't make sense to handle that in this PR then.